packages/dds/tree/src/test/simple-tree/api/snapshotCompatibilityChecker.spec.ts

```diff
@@ -0,0 +1,58 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	checkCompatibility,
+	normalizeFieldSchema,
+	importCompatibilitySchemaSnapshot,
+	SchemaFactory,
+	exportCompatibilitySchemaSnapshot,
+	TreeViewConfiguration,
+	type SchemaCompatibilityStatus,
+} from "../../../simple-tree/index.js";
+import { strict as assert } from "node:assert";
+
+describe("snapshotCompatibilityChecker", () => {
+	it("parse and snapshot can roundtrip schema", () => {
+		const factory = new SchemaFactory("test");
+		const Schema = factory.optional(factory.string, {});
+
+		const view = new TreeViewConfiguration({ schema: Schema });
+		const snapshot = exportCompatibilitySchemaSnapshot(view);
+		const parsedView = importCompatibilitySchemaSnapshot(snapshot);
+
+		const normalizedView = normalizeFieldSchema(view.schema);
+
+		assert.equal(normalizedView.allowedTypeSet.size, 1);
+		assert.equal(
+			normalizedView.allowedTypesIdentifiers.has("com.fluidframework.leaf.string"),
+			true,
+		);
+	});
+
+	it("checkCompatibility detects incompatible schemas", () => {
```

Comment:
I think you should also have a test in here somewhere showing a case which actually did schema evolution right (and thus used the data thats in view but not stored schema) like those in docs/docs/data-structures/tree/schema-evolution/types-of-changes.mdx

Specifically actually do forward schema compatibility with allowUnknownOptionalFields and/or staged schema, and check that this reports the expected compatibility. These APis were added specifically to allow checking that so we should prabably have at least one minimal test which does it (thats the whole reason we had to use view schema stuff instead of the existing stored schema APIs)